src/vic3_world/countries/vic3_country.h

```diff
@@ -71,6 +71,32 @@ class Country
    void AddInterestGroupId(int ig_id) { ig_ids_.push_back(ig_id); }
    void AddPuppet(int puppet) { puppets_.insert(puppet); }
 
+   /// <summary>
+   /// Set the world that this country is a part of. Used for stats stored in global lists (eg, country rank). Set when inserting this country into a world.
+   /// </summary>
+   /// <param name="world">world</param>
+   void SetWorld(const World& world) { world_.emplace(world); }
```

Comment:
The other option is for GetAcquiredTechnologies et al to take in a world parameter directly. This fixes the ugliness of having a SetWorld function, but it means that you now have to pass the world around as well as the country.